### PR TITLE
Add descriptions of ClusterRole and ClusterRoleBinding to docs/reference/glossary/rbac.md

### PR DESCRIPTION
**Pull request by ym1024**
### Description

The RBAC section of the glossary page explains only two types of Kubernetes objects: `Role` and `RoleBinding`.
However, RBAC utilizes four types of Kubernetes objects: `Role`, `RoleBinding`, `ClusterRole`, and `ClusterRoleBinding`.
This PR adds descriptions of `ClusterRole` and `ClusterRoleBinding` along with a link to the RBAC documentation to help readers better understand RBAC.

### Issue

Closes: #48967

### CODE REVIEW

**Comment by ym1024** [comment]
Preview Page: https://deploy-preview-49077--kubernetes-io-main-staging.netlify.app/docs/reference/glossary/?all=true#term-rbac

**Comment by reylejano** [comment]
The original context of role and role binding (lower case) did not mean to infer to the object types which are in uppercamel case of Role and RoleBinding for namespaced and ClusterRole and ClusterRoleBinding for cluster-wide
Since the change proposed in this PR may provide clarity for new users and folks unfamiliar with the style guide, this is a good change
/approve

File: content/en/docs/reference/glossary/rbac.md [hunk 1]
```diff
@@ -15,5 +15,10 @@ tags:
 
 <!--more--> 
 
-RBAC utilizes *roles*, which contain permission rules, and *role bindings*, which grant the permissions defined in a role to a set of users.
+RBAC utilizes four kinds of Kubernetes objects:
+* **Role**: Defines permission rules in a specific namespace.
+* **ClusterRole**: Defines permission rules cluster-wide.
+* **RoleBinding**: Grants the permissions defined in a role to a set of users in a specific namespace.
+* **ClusterRoleBinding**: Grants the permissions defined in a role to a set of users cluster-wide.
```
Comment (sftim) [inline content/en/docs/reference/glossary/rbac.md:22]: Here's how we'd usually style this:
```markdown
Role
: Defines permission rules in a specific namespace.
ClusterRole
: Defines permission rules cluster-wide.
RoleBinding
: Grants the permissions defined in a role to a set of users in a specific namespace.
ClusterRoleBinding
: Grants the permissions defined in a role to a set of users cluster-wide.
```

(untested, but Hugo definitely supports rendering those to a description list)

Comment (ym1024) [reply content/en/docs/reference/glossary/rbac.md]: Thanks!
The suggested style is working well with Hugo.
[Preview Page](https://deploy-preview-49077--kubernetes-io-main-staging.netlify.app/docs/reference/glossary/?all=true#term-rbac)

I've added a minor detail, we need to have a newline between each term:
```
Role
: Defines permission rules in a specific namespace.

ClusterRole
: Defines permission rules cluster-wide.

RoleBinding
: Grants the permissions defined in a role to a set of users in a specific namespace.

ClusterRoleBinding
: Grants the permissions defined in a role to a set of users cluster-wide.
```

Without a newline, it was rendered like:
スクリーンショット 2024-12-14 5 07 23

**Comment by sftim** [comment]
Thanks
/lgtm

**Pull request merged**